File: Inboxes/Python/Web/Plain/src/cache/adapters/local.py
import time
import threading
from cache.cache_adapter import Cache

class Local(Cache):
    def __init__(self):
        self._cache = {}
        self._lock = threading.Lock()

    def get(self, key):
        with self._lock:
            entry = self._cache.get(key)
            if entry and (entry["exp"] is None or entry["exp"] > time.time()):
                return entry["val"]
            if entry:
                del self._cache[key]
        return None

    def set(self, key, value, ttl=300):
        with self._lock:
            self._cache[key] = {"val": value, "exp": time.time() + ttl}

    def delete(self, key):
        with self._lock:
            self._cache.pop(key, None)

File: SemanticSearch/PHP/Web/Plain/src/Storage/Adapters/Local.php
<?php
require_once __DIR__ . '/../CacheAdapter.php';

class Local implements CacheAdapter {
    private static array $store = [];

    public function get(string $key): mixed {
        if (!isset(self::$store[$key])) return null;
        if (self::$store[$key]['exp'] < time()) {
            unset(self::$store[$key]);
            return null;
        }
        return self::$store[$key]['val'];
    }

    public function set(string $key, mixed $value, int $ttl = 300): void {
        self::$store[$key] = ['val' => $value, 'exp' => time() + $ttl];
    }

    public function delete(string $key): void {
        unset(self::$store[$key]);
    }
}